plugins/hls-eval-plugin/src/Ide/Plugin/Eval/CodeLens.hs

```diff
@@ -616,14 +620,30 @@ evals (st, fp) df stmts = do
         let opts = execOptions{execSourceFile = fp, execLineNumber = l}
          in execStmt stmt opts
 
-runGetSession :: MonadIO m => IdeState -> NormalizedFilePath -> m HscEnvEq
-runGetSession st nfp =
-    liftIO $
-        runAction "getSession" st $
-            use_
-                GhcSession
-                -- GhcSessionDeps
-                nfp
+ghcSessionDepsDefinition :: HscEnvEq -> NormalizedFilePath -> Action HscEnv
+ghcSessionDepsDefinition env file = do
+        let hsc = hscEnvWithImportPaths env
+        deps <- use_ GetDependencies file
+        let tdeps = transitiveModuleDeps deps
+        ifaces <- uses_ GetModIface tdeps
```

Comment:
Does this work? ifaces are not guaranteed to have linkables included. I feel like the ideal solution would be restarting the shake session after "marking" all transitive dependencies as needing linkables.

Reply:
We could inspect the ifaces first, and then only restart if there are missing linkables.

Reply:
It works, I think the interactive driver will generate the missing linkables. It would be better if we generated them ourselves though, in order to cache them. But I'm not sure what's the best way to "mark" things as you describe. I suspect some reorganisation of rules might be required. 

 Would you be able to take a look at that in a future PR?

Reply:
That is unclear to me too at this point, but I do plan to look at this in the future.